oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -136,7 +138,9 @@ pub fn write_log_message<T: AsRef<str>>(message: T) -> Result<(), OakStatus> {
 ///
 /// See [`tf_model_infer`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#tf_model_infer).
 #[cfg(feature = "oak-tf")]
-pub fn tf_model_infer(input_vector: &[u8]) -> Result<Inference, OakStatus> {
+pub fn tf_model_infer(
+    input_vector: &[u8],
+) -> Result<Result<Inference, TfModelInferError>, OakStatus> {
```

Comment:
Not sure what the end is exaclty.

The double result is currently motivated by: 
- the outer result is the result of the transport (hence failing to an OakStatus), 
- the inner result is the result of the extension.

I guess when we have a different solution for the transport this could change. Do you think I am missing something or we should have  a different solution here?

Reply:
Ah yes you are right, this is the SDK, my bad. I guess here we would always expect the double result then.